TelemetryEvents.swift

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+private let PrefKeySearches = "Telemetry.Searches"
+
+public class SearchTelemetryFactory {
+    // For data consistency, the strings used here are identical to the ones reported in Android.
+    public enum Source: String {
+        case URLBar = "actionbar"
+        case QuickSearch = "listitem"
+        case Suggestion = "suggestion"
+    }
+
+    private init() {}
+
+    public class func makeEvent(engineID engineID: String, source: Source) -> TelemetryEvent {
+        return SearchTelemetryEvent(engineWithSource: "\(engineID).\(source.rawValue)")
```

Comment:
I was trying to design the telemetry API to be context-agnostic. Note that recording to prefs requires a `prefs`, which we don't have in a `SearchViewController`.